crypto/asn1/a_time.c

```diff
@@ -86,12 +86,8 @@ ASN1_GENERALIZEDTIME *ASN1_TIME_to_generalizedtime(const ASN1_TIME *t,
         goto err;
     str = (char *)ret->data;
     /* Work out the century and prepend */
-    if (t->data[0] >= '5')
-        strcpy(str, "19");
-    else
-        strcpy(str, "20");
-
-    strcat(str, (const char *)t->data);
+    memcpy(str, t->data[0] >= '5' ? "19" : "20", 2);
+    OPENSSL_strlcpy(str + 2, (const char *)t->data, t->length + 1);
```

Comment:
No, t->length + 2 is allocated four lines above.  This actually allocates +3 for the terminator.
I'll add some comments to make it clearer.